apps/omg/lib/omg/state/transaction/validator/fee_claim.ex

```diff
@@ -0,0 +1,46 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.State.Transaction.Validator.FeeClaim do
+  @moduledoc """
+  Contains generic validation rules for `Transaction.Fee` transactions. Specific transaction type's validation
+  is passed to `Transaction.Protocol.can_apply?`
+  """
+
+  alias OMG.State.Core
+  alias OMG.State.Transaction
+
+  @type fee_claim_error :: :claiming_unsupported_token | :claiming_more_than_collected
+
+  @spec can_claim_fees(Core.t(), Transaction.Recovered.t()) ::
+          {:ok, %{}} | {{:error, fee_claim_error()}, Core.t()}
+  def can_claim_fees(
+        %Core{fee_claimer_address: owner, fees_paid: fees_paid} = state,
+        %Transaction.Recovered{signed_tx: %{raw_tx: fee_tx}}
+      ) do
+    # NOTE: Fee claiming transaction does not transfer funds. It spends pseudo-output resultant of fees collection
+    with outputs = make_outputs(owner, fees_paid),
```

Comment:
It is not pattern-matching. I could write this `outputs <- make_outputs(...)` but I need to be clear there is no fail-case here. However fn returns is used in the latter, so it more readable inside `with` (I've seen some of the assignment cases in the `with` in the codebase)

Reply:
Wait now... you're contradicting yourself.
Earlier you said:
```
with is a concept of railway oriented programming pattern so there is happy-path flow and the error flow. In the happy-path functions use the results of the previous executions when successful to produce the final result. The first error is returned otherwise.
```
Where is any of it now?
Why is there an else clause?

Reply:
In the above code a happy-path is a chain of functions:  `make_outputs` & `can_apply?` and 1st will not fail which is explicitly show by using `=`. Final result is implicit result of prev functions successful execution.

I would like to avoid `with-else` as much as possible. Here is simple wrapping error so I think is justified.